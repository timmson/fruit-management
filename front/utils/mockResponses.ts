export function okAndJson(json: any): Promise<Response> {
    return Promise.resolve({
        arrayBuffer(): Promise<ArrayBuffer> {
            return Promise.resolve(undefined);
        },
        blob(): Promise<Blob> {
            return Promise.resolve(undefined);
        },
        body: undefined,
        bodyUsed: false,
        clone(): Response {
            return undefined;
        },
        formData(): Promise<FormData> {
            return Promise.resolve(undefined);
        },
        headers: undefined,
        json(): Promise<any> {
            return Promise.resolve(json);
        },
        ok: true,
        redirected: false,
        status: 200,
        statusText: "OK",
        text(): Promise<string> {
            return Promise.resolve("");
        },
        type: undefined,
        url: ""
    })
}