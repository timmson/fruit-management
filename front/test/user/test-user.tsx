import React from "react"
import renderer from "react-test-renderer"
import User from "../../src/user/user"
import Context from "../../src/context"
import {okAndJson} from "../../utils/mockResponses"

global.fetch = jest.fn(() => okAndJson({}))

describe("User should", () => {

	test("return user form", () => {
		const context = [{user: {samaccountname: "user"}}]

		const component = renderer.create(
			<Context.Provider value={context}>
				<User/>
			</Context.Provider>
		)

		expect(component.toJSON()).toMatchSnapshot()

		component.unmount()
	})

})
