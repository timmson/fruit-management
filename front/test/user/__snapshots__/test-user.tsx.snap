// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should return user form 1`] = `
[
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h3>
        Пользователь
      </h3>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col"
    >
      <input
        name="user"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="user"
      />
    </div>
  </div>,
  <div
    className="row mt-3"
  >
    <div
      className="col"
    >
      <h3>
        Мои задачи
      </h3>
      <table
        className="basic w-100"
      >
        <thead>
          <tr
            className="fw-bold"
          >
            <td>
              Задача
            </td>
            <td>
              Состояние
            </td>
            <td>
              %
            </td>
            <td>
              Приоритет
            </td>
            <td>
              Статус
            </td>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  <div
    className="row mt-3"
  >
    <div
      className="col"
    >
      <h3>
        Моя активность
      </h3>
      <table
        className="basic w-100"
      >
        <thead>
          <tr
            className="fw-bold"
          >
            <th>
              Задача
            </th>
            <th>
              Статус
            </th>
            <th>
              ПН
            </th>
            <th>
              ВТ
            </th>
            <th>
              СР
            </th>
            <th>
              ЧТ
            </th>
            <th>
              ПТ
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className="fw-bold"
          >
            <td
              className="text-end"
              colSpan="2"
            >
              Итого
            </td>
            <td
              className="text-center"
            >
              0
            </td>
            <td
              className="text-center"
            >
              0
            </td>
            <td
              className="text-center"
            >
              0
            </td>
            <td
              className="text-center"
            >
              0
            </td>
            <td
              className="text-center"
            >
              0
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
]
`;
